lib/galaxy/tool_util/cwl/parser.py

```diff
@@ -56,29 +64,42 @@
 SUPPORTED_WORKFLOW_REQUIREMENTS = SUPPORTED_TOOL_REQUIREMENTS + [
 ]
 
+PERSISTED_REPRESENTATION = "cwl_tool_object"
 
-def tool_proxy(tool_path=None, tool_object=None, strict_cwl_validation=True):
+
+def tool_proxy(tool_path=None, tool_object=None, strict_cwl_validation=True, tool_directory=None, uuid=None):
     """ Provide a proxy object to cwltool data structures to just
     grab relevant data.
     """
     ensure_cwltool_available()
-    tool = to_cwl_tool_object(
+    # if uuid is None:
+    #    raise Exception("tool_proxy must be called with non-None uuid")
+    tool = _to_cwl_tool_object(
         tool_path=tool_path,
         tool_object=tool_object,
-        strict_cwl_validation=strict_cwl_validation
+        strict_cwl_validation=strict_cwl_validation,
+        tool_directory=tool_directory,
+        uuid=uuid
     )
     return tool
 
 
-def tool_proxy_from_persistent_representation(persisted_tool, strict_cwl_validation=True):
+def tool_proxy_from_persistent_representation(persisted_tool, strict_cwl_validation=True, tool_directory=None):
+    """Load a ToolProxy from a previously persisted representation."""
     ensure_cwltool_available()
-    tool = to_cwl_tool_object(persisted_tool=persisted_tool, strict_cwl_validation=strict_cwl_validation)
+    if PERSISTED_REPRESENTATION == "cwl_tool_object":
```

Comment:
It is an semi-open question in the CWL branch what the persisted representation should be. I think the current setting is right and once the plumbing is all merged and these things work - it will probably be this way - but I'm hesitant to commit to a path until the whole thing is ready to go and working.